src/lib/input/input.ts

```diff
@@ -229,6 +234,15 @@ export class MdInput implements ControlValueAccessor, AfterContentInit, OnChange
 
   @ViewChild('input') _inputElement: ElementRef;
 
+  _elementType: 'input' | 'textarea';
+
+  constructor(public _elementRef: ElementRef) {
```

Comment:
It can just be
```ts
 constructor(elementRef: ElementRef) {
```
Adding a `public` or `private` modifier here makes it a field on the class which isn't necessary.